compiler/aarch64/codegen/TreeEvaluatorTable.hpp

```diff
@@ -0,0 +1,748 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+/*
+ * This table is #included in a static table.
+ * Only Function Pointers are allowed.
+ */
+    TR::TreeEvaluator::BadILOpEvaluator ,		// illegal op hopefully help with uninitialized nodes
```

Comment:
Rather than using comments describing the operation, can you actually list the IL OpCode (perhaps in addition to the comment) like the other architectures do.  It makes it easier to index into this table by searching for the opcode you're interested in.  At some point, there may not be a 1-1 mapping between IL evaluator names and IL opcodes (e.g., you may create an "integerAddEvaluator" that handles multiple sizes of inputs that is shared across multiple IL opcodes) and in that case its easier to search the table for the IL opcode/evaluator pairing by using the IL opcode name.

Reply:
this was a cleaner hack to get it to build. Otherwise it fails static assertion (nb of IL op == nb of Ops in tree evaluator table)
I could just ifndef (TR_TARGET_AARCH64) the static assertion rather and loose the tree evaluator stuff and reassess later so this gets merged in and less bloat on the commit? thoughts ? @0xdaryl